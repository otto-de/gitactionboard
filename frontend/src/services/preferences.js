import storageService from "@/services/storageService";

class Preferences {
  __get__(configKey) {
    return storageService.getItem(configKey);
  }

  __set__(configKey, value) {
    storageService.setItem(configKey, value);
  }

  get showHealthyBuilds() {
    const showHealthBuildConfig = this.__get__("show-healthy-builds");
    return !(showHealthBuildConfig && showHealthBuildConfig === "false");
  }

  set showHealthyBuilds(value) {
    this.__set__("show-healthy-builds", value);
  }

  get disableIdleOptimization() {
    return this.__get__("disable-max-idle-time") === "true";
  }

  set disableIdleOptimization(value) {
    this.__set__("disable-max-idle-time", value);
  }

  get maxIdleTime() {
    const maxIdleTimeConfig = this.__get__("max-idle-time");
    return maxIdleTimeConfig ? parseInt(maxIdleTimeConfig) : 5;
  }

  set maxIdleTime(value) {
    this.__set__("max-idle-time", value);
  }

  get hiddenElements() {
    return JSON.parse(this.__get__("hidden-elements") || "{}");
  }

  set hiddenElements(value) {
    this.__set__("hidden-elements", JSON.stringify(value));
  }
}

export default new Preferences();
