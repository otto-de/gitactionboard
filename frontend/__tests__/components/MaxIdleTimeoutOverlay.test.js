import { describe, expect, it, beforeEach, afterEach, vi } from 'vitest';
import { mount } from '../test-utils';

import MaxIdleTimeoutOverlay from '@/components/MaxIdleTimeoutOverlay.vue';
import preferences from '@/services/preferences';

describe('<MaxIdleTimeoutOverlay />', () => {
  const { reload } = window.location;

  beforeEach(() => {
    vi.spyOn(preferences, 'theme', 'get').mockReturnValueOnce('light');
    Object.defineProperty(window, 'location', {
      writable: true,
      value: { reload: vi.fn() }
    });
  });

  afterEach(() => {
    window.location.reload = reload;
  });

  it('should render overlay', () => {
    expect(mount(MaxIdleTimeoutOverlay).html()).toMatchSnapshot();
  });

  it('should reload page when user clicks on reload button', async () => {
    const vueWrapper = mount(MaxIdleTimeoutOverlay);

    await vueWrapper.find('[test-id="reload-button"]').trigger('click');

    expect(window.location.reload).toHaveBeenCalledOnce();
  });
});
