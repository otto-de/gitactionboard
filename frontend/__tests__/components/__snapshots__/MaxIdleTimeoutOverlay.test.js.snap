// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<MaxIdleTimeoutOverlay /> > should render overlay 1`] = `
<!--teleport start-->
<div aria-modal="true" class="v-overlay v-overlay--active v-theme--light v-locale--is-ltr v-dialog" role="dialog" style="z-index: 2400;">
  <transition-stub appear="true" css="true" name="fade-transition" persisted="false">
    <div class="v-overlay__scrim"></div>
  </transition-stub>
  <transition-stub appear="true" css="true" name="dialog-transition" persisted="true">
    <div class="v-overlay__content" style="width: auto;">
      <div class="v-card v-theme--light v-card--density-default v-card--variant-elevated pa-3 rounded-lg">
        <!---->
        <div class="v-card__loader">
          <div aria-hidden="true" aria-valuemax="100" aria-valuemin="0" class="v-progress-linear v-theme--light v-locale--is-ltr" role="progressbar" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);">
            <!---->
            <div class="v-progress-linear__background" style="width: 100%;"></div>
            <transition-stub appear="false" css="true" name="fade-transition" persisted="false">
              <div class="v-progress-linear__indeterminate">
                <div class="v-progress-linear__indeterminate long"></div>
                <div class="v-progress-linear__indeterminate short"></div>
              </div>
            </transition-stub>
            <!---->
          </div>
        </div>
        <!---->
        <!---->
        <div class="v-alert v-theme--light bg-warning v-alert--density-default v-alert--variant-flat" role="alert">
          <!---->
          <span class="v-alert__underlay"></span>
          <!---->
          <div class="v-alert__prepend">
            <i aria-hidden="true" class="mdi-alert-circle mdi v-icon notranslate v-theme--light" density="default" style="font-size: 28px; height: 28px; width: 28px;"></i>
          </div>
          <div class="v-alert__content">
            <div class="v-alert-title">Stopped automatic page refreshment due to maximum idle timeout</div>
            <!---->
            <!---->
          </div>
          <!---->
          <!---->
        </div>
        <div class="v-card-actions pt-10">
          <div class="v-spacer"></div>
          <button class="v-btn v-btn--slim v-theme--light v-btn--density-default elevation-3 v-btn--size-default v-btn--variant-text" test-id="reload-button" type="button">
            <span class="v-btn__overlay"></span>
            <span class="v-btn__underlay"></span>
            <span class="v-btn__prepend">
              <i aria-hidden="true" class="mdi-refresh mdi v-icon notranslate v-theme--light v-icon--size-default"></i>
            </span>
            <span class="v-btn__content" data-no-activator> Reload Page </span>
            <!---->
            <!---->
          </button>
          <div class="v-spacer"></div>
        </div>
        <!---->
        <!---->
        <span class="v-card__underlay"></span>
      </div>
    </div>
  </transition-stub>
</div>
<!--teleport end-->
`;
